src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -27,12 +28,52 @@ protected override Task<ImmutableArray<ISymbol>> GetSymbolsAsync(SyntaxContext c
                 context.SemanticModel, position, context.Workspace, options, cancellationToken);
         }
 
+        internal async override Task<bool> ShouldProvidePreselectedItemsAsync(CompletionContext completionContext, SyntaxContext syntaxContext, Document document, int position, Lazy<ImmutableArray<ITypeSymbol>> inferredTypes, OptionSet options)
+        {
+            var sourceText = document.GetTextSynchronously(CancellationToken.None);
+            if (ShouldTriggerInArgumentLists(sourceText, options))
+            {
+                if (completionContext.Trigger.Kind == CompletionTriggerKind.Insertion &&
+                    await IsTriggerInArgumentListAsync(document, position - 1, CancellationToken.None).ConfigureAwait(false) == true)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
         protected override bool IsInstrinsic(ISymbol s)
             => s is ITypeSymbol ts && ts.IsIntrinsicType();
 
         internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
         {
-            return CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
+            return ShouldTriggerInArgumentLists(text, options)
+                ? CompletionUtilities.IsTriggerCharacterOrArgumentListCharacter(text, characterPosition, options)
+                : CompletionUtilities.IsTriggerCharacter(text, characterPosition, options);
+        }
+
+        internal async override Task<bool> IsSyntacticTriggerCharacterAsync(Document document, int caretPosition, CompletionTrigger trigger, OptionSet options, CancellationToken cancellationToken)
+        {
+            if (trigger.Kind == CompletionTriggerKind.Insertion)
+            {
+                var result = await IsTriggerOnDotAsync(document, caretPosition - 1, cancellationToken).ConfigureAwait(false);
+                if (result.HasValue)
+                {
+                    return result.Value;
+                }
```

Comment:
do we even need this check?  i feel like we just need the Argument-List check, where ew might override the intiial 'maybe' check.  it's the only thing that can make us return false.  otherwise, we fall through and return true.

Reply:
I liked the pattern established by IsTriggerOnDot where it only answers the question if it's confident, and then having the explicit fall through true return is clearer about why it's returning true (if I add a comment) and leaves a clear pattern if anyone else has to come in and add a 3rd one of these checks.

Reply:
Going to keep this unless you have more objections.